cupyx/scipy/sparse/_data.py

```diff
@@ -121,15 +121,18 @@ def power(self, n, dtype=None):
 
 
 def _find_missing_index(ind, n):
-    for k, a in enumerate(ind):
-        if k != a:
-            return k
+    positions = cupy.arange(ind.size)
+    diff = ind != positions
+    non_zero = cupy.argwhere(diff).ravel()
```

Comment:
`cupy.argwhere` also causes device-to-host sync because the length of return values varies depending on the value of elements of the inputs.

How about the following implementation?

```py
return cupy.where(
    diff.any(),
    diff.argmax(),
    cupy.asarray(ind.size if ind.size < n else -1))
```
```